File: Napakalaki/src/napakalaki/Napakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author francisco
 */
public class Napakalaki {
    private int currentPlayerIndex;
    private static final Napakalaki instance = new Napakalaki();
    private Monster currentMonster;
    private CardDealer dealer;
    private Player currentPlayer;
    
    private ArrayList<Player> players = new ArrayList();
    
    
    private Napakalaki(){}
    
    private void initPlayers(ArrayList<String> names){}
    
    private Player nextPlayer(){}
    
    private boolean nextTurnAllowed(){}
    
    public static Napakalaki getInstance(){
        return instance;
    }
    
    public CombatResult developCombat(){}
    
    public void discardVisibleTreasures(ArrayList<Treasure> treasures){}
    
    public void discardHiddenTreasures(ArrayList<Treasure> treasures){}
    
    public void makeTreasuresVisibles(ArrayList<Treasure> treasures){}
    
    public void initGame(ArrayList<String> players){}
    
    public Player getCurrentPlayer(){
        return currentPlayer;
    }
    
    public Monster getCurrentMonster(){
        return currentMonster;
    }
    
    public boolean nextTurn(){}
    
    public boolean endOfGame(CombatResult result){}
}

File: Napakalaki_Ruby/napakalaki.rb
#encoding: utf-8

require 'singleton'
require_relative 'card_dealer.rb'

module NapakalakiGame
  class Napakalaki
    include Singleton

    attr_reader :currentPlayer, :currentMonster

    def initialize
      @currentPlayerIndex = 0
      @currentPlayer = nil
      @currentMonster = nil
      @players = Array.new
      @dealer = CardDealer.instance
    end

    def initPlayers (names)
    end

    def nextPlayer
    end

    def nextTurnAllowed
    end

    def developCombat
    end

    def discardVisibleTreasures (treasures)
    end

    def discardHiddenTreasures (treasures)
    end

    def makeTreasuresVisibles (treasures)
    end

    def initGame (players)
    end

    def nextTurn
    end

    def endOfGame (result)
    end

    private :initPlayers, :nextPlayer, :nextTurnAllowed
  end
end
